### PR TITLE
Daffio guide docs navigation

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
The guide docs sidebar in daff.io isn't organized very well and is basically unreadable.

## What is the new behavior?
- The guide docs sidebar is organized with nested accordions.
- The directory/file structure under `docs` is better organized. Previously, there were components under `docs/shared` which only pertained to the guide docs, so all of that was moved under a new `docs/guide-docs` directory.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information
More work will probably need to be done on the accordions in the future to allow for a slimmer accordion-item. Right now, they look a little big with nested accordions.

### CODE REVIEW

**Comment by lderrickable** [comment]
https://github.com/graycoreio/daffodil/pull/1555 should be reviewed first

**Review by griest024 — CHANGES_REQUESTED** [review]
The ToC links in guides don't work but I assume you'll fix that in the ToC PR?

File: apps/daffio/src/app/docs/guide-docs/components/guide-viewer/guide-viewer.component.ts
Comment (lderrickable) [reply apps/daffio/src/app/docs/guide-docs/components/guide-viewer/guide-viewer.component.ts:14]: In the future this will also have the table of contents as a sidebar on the right, while the API docs will not

File: apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.html
Comment (lderrickable) [reply apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.html]: I got rid of the root component and I moved `daff-accordion` into the sidebar. I removed the accordion from inside the guides-list component too because in the future (once I implement the nesting accordion stuff) there should only be one accordion (the accordion-items will derive their indent state from their parent accordion).

File: libs/design/src/molecules/accordion/accordion/accordion.component.html
Comment (griest024) [inline libs/design/src/molecules/accordion/accordion/accordion.component.html:1]: I think this is probably fine but I wonder if @xelaint has thoughts about the implications of this change.

Comment (lderrickable) [reply libs/design/src/molecules/accordion/accordion/accordion.component.html]: I removed this change from this PR

**Comment by lderrickable** [comment]
There are definitely some issues with the way the sidebar is still, but I think they are all either because I need to do some updates to the README files in the packages themselves or because I need to fix a bug in the `daff-nav-accordion-item`.

File: apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts
Comment (damienwebdev) [inline apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts:9]: I think this component is more appropriately named `guide-nav`.

File: apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts
Comment (damienwebdev) [inline apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts:11]: needs `OnPush`

File: apps/daffio/src/app/docs/guide-docs/views/guide-view.component.ts
Comment (damienwebdev) [inline apps/daffio/src/app/docs/guide-docs/views/guide-view.component.ts:14]: After looking at the API view and the guide view, they look the same, can these be centralized instead?

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-trie.ts
Comment (damienwebdev) [inline tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-trie.ts:120]: Doesn't need changed.

File: apps/daffio/src/app/core/sidebar/sidebar.module.ts
Comment (damienwebdev) [inline apps/daffio/src/app/core/sidebar/sidebar.module.ts:23]: This Accordion import isn't needed.

**Review by damienwebdev — CHANGES_REQUESTED** [review]

File: apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts
Comment (lderrickable) [reply apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts]: changed

File: apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts
Comment (lderrickable) [reply apps/daffio/src/app/docs/guide-docs/components/guides-list/guides-list.component.ts]: added

File: tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-trie.ts
Comment (lderrickable) [reply tools/dgeni/src/transforms/daffodil-guides-package/helpers/navigation-trie.ts:120]: I think I can only fix this with a squash commit, because git is just being weird with its detection of changes. I can do this after you add a "lgtm" and before you approve.

File: apps/daffio/src/app/core/sidebar/sidebar.module.ts
Comment (lderrickable) [reply apps/daffio/src/app/core/sidebar/sidebar.module.ts]: removed

**Comment by damienwebdev** [comment]
Needs a rebase, but otherwise, LGTM

File: apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html
Comment (xelaint) [inline apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html:3]: All the h3's should be changed to divs.

**Review by xelaint — CHANGES_REQUESTED** [review]

File: apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html
Comment (xelaint) [inline apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html:1]: Doesn't need to be here. This section should just be scrollable.

**Review by xelaint — CHANGES_REQUESTED** [review]

File: apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html
Comment (lderrickable) [reply apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html:1]: Do you mean that the `daff-accordion` doesn't need to be there or that the `*ngIf` doesn't need to be there or both?

Comment (xelaint) [reply apps/daffio/src/app/docs/guide-docs/components/guide-nav/guide-nav.component.html:1]: Nvm, I was confused with the structure. This can stay.

**Comment by xelaint** [comment]
@damienwebdev Are we going to merge this and then make updates to guides-nav after `daff-tree` is complete?

**Comment by damienwebdev** [comment]
@xelaint yes.

**Review by damienwebdev — APPROVED** [review]

**Review by xelaint — APPROVED** [review]

**Review by griest024 — APPROVED** [review]

**Pull request merged**
